### PR TITLE
Add nix build for acvm-simulator

### PR DESCRIPTION
**Pull request by jonybur**
- Runs acvm-simulator nix build and uploads artifacts

Linked to: https://github.com/noir-lang/acvm-simulator/pull/19

### CODE REVIEW

**Comment by jonybur** [comment]
This is working. Needs rolling back so it runs on schedule and so that it pushes to the same branch instead of testing-branch.

This needs to be rolled back:

https://github.com/noir-lang/acvm-simulator-wasm/pull/1/commits/4f102b4498aaaf6e8d4c9f4c97c9987f3c0edb95

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -43,14 +44,20 @@ jobs:
         with:
           nix_path: nixpkgs=channel:nixos-22.11
 
-      # - uses: cachix/cachix-action@v12
-      #   with:
-      #     name: acvm-simulator-cache
-      #     authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      - name: Build with Nix
+        run: |
+          nix build github:noir-lang/acvm-simulator/jb-nix-flake
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: `jb-nix-flake` should branch be taken from environment?

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -43,14 +44,20 @@ jobs:
...
         run: |
-          nix-shell acvm-simulator.nix --pure --run "./build.sh"
+          realpath ./result > result_path.txt
+          echo "RESULT_PATH=$(cat result_path.txt)" >> $GITHUB_ENV
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: release-artifact
+          path: ${{ env.RESULT_PATH }}
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: would symlink not work here? if not would `RESULT_PATH=$(readlink result)` be easier?

File: .github/workflows/update.yml (hunk 1 above)
Comment (jonybur) [reply .github/workflows/update.yml]: Yep this is just for testing purposes

File: .github/workflows/update.yml (hunk 2 above)
Comment (jonybur) [reply .github/workflows/update.yml]: Fixed

File: .github/workflows/update.yml [hunk 3]
```diff
@@ -6,8 +6,9 @@ on:
       acvm-simulator-ref:
         description: The ACVM simulator reference to checkout
         required: false
-  schedule:
-    - cron: "0 2 * * *" # run at 2 AM UTC
+  push:
+    branches:
```
Comment (kobyhallx) [inline .github/workflows/update.yml:10]: Thise should be removed before merge?

File: .github/workflows/update.yml [hunk 4]
```diff
@@ -6,8 +6,9 @@ on:
...
+      - jb-nix-flake
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: Thise should be removed before merge?

File: .github/workflows/update.yml [hunk 5]
```diff
@@ -59,9 +65,12 @@ jobs:
...
+          BRANCH_NAME="testing-branch-$BRANCH_SUFFIX"
+          git checkout -B $BRANCH_NAME
           git add .
-          git commit -m "tracking acvm-simulator@${{ steps.collect-rev.outputs.NOIR_REV_SHORT }}"
-          git push --force
+          git commit -m "tracking acvm-simulator@${{ env.SIMULATOR_REV_SHORT }}"
+          git push origin $BRANCH_NAME
 
       # - name: Setup NodeJS
       #   uses: actions/setup-node@v3
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: NodeJS setup not needed

File: .github/workflows/update.yml [hunk 6]
```diff
@@ -59,9 +65,12 @@ jobs:
...
       - name: Commit updates
         run: |
+          BRANCH_SUFFIX=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 13 ; echo '')
+          BRANCH_NAME="testing-branch-$BRANCH_SUFFIX"
+          git checkout -B $BRANCH_NAME
           git add .
-          git commit -m "tracking acvm-simulator@${{ steps.collect-rev.outputs.NOIR_REV_SHORT }}"
-          git push --force
+          git commit -m "tracking acvm-simulator@${{ env.SIMULATOR_REV_SHORT }}"
+          git push origin $BRANCH_NAME
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: Are we not suppose to push to this repo instead of `origin`?

Comment (jonybur) [reply .github/workflows/update.yml]: This was also just for testing (so that it pushes to a testing branch)

**Review by kobyhallx — APPROVED** [review]
+1

**Pull request merged**